src/lib/APS2.cpp

```diff
@@ -487,58 +528,98 @@ void APS2::write_memory(const uint32_t & addr, const vector<uint32_t> & data){
 	 * data = vector<uint32_t> data
 	 */
 
-	if (data.size() == 0) {
-		return;
-	}
```

Comment:
Hmmm, we should add coverage because I think it will break the new firmware too.   0 bytes to transfer DataMover commands are illegal.